### DISCUSSION TITLE
Proposal: attributes on type parameters

### DISCUSSION THREAD

**Discussion by hez2010** [dotnet/csharplang#2716] [Language Ideas]
### Proposal
Allow attributes on type parameters.

### Usage
```csharp
public async Task<ValueTuple<bool, [MaybeNull]T> GetAsync<T>();
```

note:
It's not
```csharp
public async Task<ValueTuple<bool, T> GetAsync<[MaybeNull]T>();
```

**Comment by Joe4evr** [comment]
[This already works.](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFAG0CCAXDAnAlgIwFcMBTAVQGcBDAcxIApMcDiSAVK7OjCgOgHESAOxJ4AxgAVOVALYlS2AJQBdZFADMAAliaAsgE8AwoQoYA9jKZ4ipTSE1WWpZAG9kAX2RekG7TE2GADwoBsamFsqabAB8rh5AA===)

**Comment by hez2010** [comment]
@Joe4evr thanks! However [MaybeNull] could not be used on GenericParameter

**Comment by Joe4evr** [comment]
Oh, I see what you're doing now. You can only apply attributes on generic parameters at the place where the *type* is declared, meaning you'd have to own it. But even that won't help for a nested generic argument like this.

This is quite a conundrum. Pinging @jcouv for insight.
